File: framework/ecommerce-framework-dxa-module/src/main/java/com/sdl/ecommerce/dxa/model/ECommerceCategoryReference.java
package com.sdl.ecommerce.dxa.model;

import com.sdl.ecommerce.api.model.Category;
import com.sdl.ecommerce.api.model.ECommerceEclItem;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticEntity;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticProperty;
import com.sdl.webapp.common.api.model.entity.AbstractEntityModel;

import static com.sdl.webapp.common.api.mapping.semantic.config.SemanticVocabulary.SDL_CORE;

/**
 * ECommerce Category Reference
 *
 * @author nic
 */
@SemanticEntity(entityName = "ECommerceCategory", vocabulary = SDL_CORE, prefix = "e", public_ = false)
public class ECommerceCategoryReference extends AbstractEntityModel {

    @SemanticProperty("e:categoryPath")
    private String categoryPath;

    @SemanticProperty("e:categoryRef")
    private ECommerceEclItem categoryRef;

    @SemanticProperty("e:categoryId")
    private String categoryId;

    // Data set by controllers
    //
    private Category category;
    private String categoryUrl;

    public String getCategoryPath() {
        return categoryPath;
    }

    public ECommerceEclItem getCategoryRef() {
        return categoryRef;
    }

    public String getCategoryId() {
        return categoryId;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

    public String getCategoryUrl() {
        return categoryUrl;
    }

    public void setCategoryUrl(String categoryUrl) {
        this.categoryUrl = categoryUrl;
    }
}

File: dxa.net/ecommerce-framework-dotnet/SDL.ECommerce.DXA/Models/ECommerceCategoryReference.cs
using Sdl.Web.Common.Models;
using SDL.ECommerce.Api.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SDL.ECommerce.DXA.Models
{
    [SemanticEntity(Vocab = CoreVocabulary, EntityName = "ECommerceCategory", Prefix = "e")]
    public class ECommerceCategoryReference : EntityModel
    {
        [SemanticProperty("e:categoryPath")]
        public string CategoryPath { get; set; }

        [SemanticProperty("e:categoryRef")]
        public ECommerceEclItem CategoryRef { get; set; }

        [SemanticProperty("e:categoryId")]
        public string CategoryId { get; set; }

        [SemanticProperty(IgnoreMapping = true)]
        public ICategory Category { get; set; }
    }
}
